{"priority": "trivial", "kind": "bug", "repository": {"links": {"self": {"href": "data/repositories/hoffmanlab/segway.json"}, "html": {"href": "#!/hoffmanlab/segway"}, "avatar": {"href": "data/bytebucket.org/ravatar/{68bf38bf-f0c1-4c9c-909e-0ee4b2c26257}ts=python"}}, "type": "repository", "name": "segway", "full_name": "hoffmanlab/segway", "uuid": "{68bf38bf-f0c1-4c9c-909e-0ee4b2c26257}"}, "links": {"attachments": {"href": "data/repositories/hoffmanlab/segway/issues/51/attachments_page=1.json"}, "self": {"href": "data/repositories/hoffmanlab/segway/issues/51.json"}, "watch": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/segway/issues/51/watch"}, "comments": {"href": "data/repositories/hoffmanlab/segway/issues/51/comments_page=1.json"}, "html": {"href": "#!/hoffmanlab/segway/issues/51/when-setting-a-mem-usage-smaller-than-10mb"}, "vote": {"href": "https://api.bitbucket.org/2.0/repositories/hoffmanlab/segway/issues/51/vote"}}, "reporter": {"display_name": "Eric Roberts", "uuid": "{cd8c1fe0-28ca-45fb-8ef9-48090b42bb80}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D"}, "html": {"href": "https://bitbucket.org/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/17548d71dfb29d015b880f48cfc01ca3d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsER-5.png"}}, "nickname": "ericr86", "type": "user", "account_id": "557058:c80ca578-03a1-4ac6-b3ee-50372a3fceee"}, "title": "When setting a mem-usage smaller than 10MB, segway will fail to run regardless of input", "component": null, "votes": 0, "watches": 1, "content": {"raw": "When trying to run segway with a small memory progression, the `ulimit` running in segway-wrapper.sh is given a negative value and will produce an \"invalid argument\" error.\r\n\r\nThis is because in `calc_mem_limit` inside segway/cluster/common.py, the given memory is subtracted by `MEM_GUARD` without a bounds check. `MEM_GUARD` is preset to 10 MB.", "markup": "markdown", "html": "<p>When trying to run segway with a small memory progression, the <code>ulimit</code> running in segway-wrapper.sh is given a negative value and will produce an \"invalid argument\" error.</p>\n<p>This is because in <code>calc_mem_limit</code> inside segway/cluster/common.py, the given memory is subtracted by <code>MEM_GUARD</code> without a bounds check. <code>MEM_GUARD</code> is preset to 10 MB.</p>", "type": "rendered"}, "assignee": {"display_name": "Eric Roberts", "uuid": "{cd8c1fe0-28ca-45fb-8ef9-48090b42bb80}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D"}, "html": {"href": "https://bitbucket.org/%7Bcd8c1fe0-28ca-45fb-8ef9-48090b42bb80%7D/"}, "avatar": {"href": "data/secure.gravatar.com/avatar/17548d71dfb29d015b880f48cfc01ca3d=httpsavatar-management--avatars.us-west-2.prod.public.atl-paas.netinitialsER-5.png"}}, "nickname": "ericr86", "type": "user", "account_id": "557058:c80ca578-03a1-4ac6-b3ee-50372a3fceee"}, "state": "wontfix", "version": null, "edited_on": null, "created_on": "2015-11-18T20:27:31.631370+00:00", "milestone": null, "updated_on": "2016-03-07T17:01:08.749717+00:00", "type": "issue", "id": 51}